crates/feldera-types/src/transport/iceberg.rs

```diff
@@ -0,0 +1,354 @@
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fmt::Display};
+use utoipa::ToSchema;
+
+/// Iceberg table read mode.
+///
+/// Three options are available:
+///
+/// * `snapshot` - read a snapshot of the table and stop.
+///
+/// * `follow` - continuously ingest changes to the table, starting from a specified snapshot
+///   or timestamp.
+///
+/// * `snapshot_and_follow` - read a snapshot of the table before switching to continuous ingestion
+///   mode.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+pub enum IcebergIngestMode {
+    /// Read a snapshot of the table and stop.
+    #[serde(rename = "snapshot")]
+    Snapshot,
+
+    /// Follow the changelog of the table, only ingesting changes (new and deleted rows).
+    #[serde(rename = "follow")]
+    Follow,
+
+    /// Take a snapshot of the table before switching to the `follow` mode.
+    #[serde(rename = "snapshot_and_follow")]
+    SnapshotAndFollow,
+}
+
+impl Display for IcebergIngestMode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            IcebergIngestMode::Snapshot => f.write_str("snapshot"),
+            IcebergIngestMode::Follow => f.write_str("follow"),
+            IcebergIngestMode::SnapshotAndFollow => f.write_str("snapshot_and_follow"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+pub enum IcebergCatalogType {
+    #[serde(rename = "rest")]
+    Rest,
+    #[serde(rename = "glue")]
+    Glue,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+pub struct GlueCatalogConfig {
```

Comment:
what is _glue_
maybe deserves a top-level struct comment?

Reply:
Glue is a "serverless data integration" solution from AWS. If you don't know what that means, you're not alone. The relevant part is, it's required to work with Iceberg tables managed by AWS.